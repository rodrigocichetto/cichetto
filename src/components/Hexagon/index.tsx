import { Icon } from 'components';

import { ThemeColors, AvailableIcons } from 'types/global';

import * as S from './styles';

export type HexagonProps = {
  icon: AvailableIcons;
  color?: ThemeColors;
  background?: ThemeColors;
};

const Hexagon = ({
  icon,
  color = 'white',
  background = 'secondary'
}: HexagonProps) => (
  <S.Wrapper background={background}>
    <S.IconWrapper>
      <Icon icon={icon} color={color} size="xlarge" />
    </S.IconWrapper>
    <S.HexagonWrapper>
      <svg
        version="1.1"
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 86.60254037844386 100"
      >
        <path
          fill="currentColor"
          d="M34.64101615137754 4.999999999999999Q43.30127018922193 0 51.96152422706632 4.999999999999999L77.94228634059948 20Q86.60254037844386 25 86.60254037844386 35L86.60254037844386 65Q86.60254037844386 75 77.94228634059948 80L51.96152422706632 95Q43.30127018922193 100 34.64101615137754 95L8.660254037844387 80Q0 75 0 65L0 35Q0 25 8.660254037844387 20Z"
        ></path>
      </svg>
    </S.HexagonWrapper>
  </S.Wrapper>
);

export default Hexagon;
