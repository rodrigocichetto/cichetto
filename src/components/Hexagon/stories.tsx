import { Story, Meta } from '@storybook/react/types-6-0';
import Hexagon from '.';

export default {
  title: 'Hexagon',
  component: Hexagon
} as Meta;

export const Default: Story = args => <Hexagon icon={args.icon} {...args} />;

Default.args = {
  icon: 'github'
};
