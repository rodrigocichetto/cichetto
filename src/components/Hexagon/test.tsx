import theme from 'styles/theme';

import { renderWithTheme } from 'utils/tests/helpers';

import Hexagon from '.';

describe('<Hexagon />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<Hexagon icon="github" />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with a different color', () => {
    const { container } = renderWithTheme(
      <Hexagon icon="github" color="secondary" background="primary" />
    );

    expect(container.firstChild).toHaveStyle({
      color: theme.colors.primary
    });

    expect(
      container.querySelector('svg[data-icon="github"]')?.parentElement
    ).toHaveStyle({
      color: theme.colors.secondary
    });
  });
});
