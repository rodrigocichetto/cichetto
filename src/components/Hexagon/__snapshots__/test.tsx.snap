// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hexagon /> should render correctly 1`] = `
.c3 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: #FAFAFA;
  font-size: 2.4rem;
}

.c2 svg {
  height: 2.4rem;
}

.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #424656;
}

.c4 {
  display: contents;
}

.c4 svg {
  -webkit-filter: drop-shadow(rgba(0,0,0,0.3) 0 0.1rem 0.15rem);
  filter: drop-shadow(rgba(0,0,0,0.3) 0 0.1rem 0.15rem);
  height: 5rem;
}

.c1 {
  position: absolute;
  z-index: 10;
  width: 100% !important;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
      color="white"
    >
      <svg
        aria-hidden="true"
        class="c3"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M12.026 2c-5.509 0-9.974 4.465-9.974 9.974 0 4.406 2.857 8.145 6.821 9.465.499.09.679-.217.679-.481 0-.237-.008-.865-.011-1.696-2.775.602-3.361-1.338-3.361-1.338-.452-1.152-1.107-1.459-1.107-1.459-.905-.619.069-.605.069-.605 1.002.07 1.527 1.028 1.527 1.028.89 1.524 2.336 1.084 2.902.829.091-.645.351-1.085.635-1.334-2.214-.251-4.542-1.107-4.542-4.93 0-1.087.389-1.979 1.024-2.675-.101-.253-.446-1.268.099-2.64 0 0 .837-.269 2.742 1.021a9.582 9.582 0 012.496-.336 9.554 9.554 0 012.496.336c1.906-1.291 2.742-1.021 2.742-1.021.545 1.372.203 2.387.099 2.64.64.696 1.024 1.587 1.024 2.675 0 3.833-2.33 4.675-4.552 4.922.355.308.675.916.675 1.846 0 1.334-.012 2.41-.012 2.737 0 .267.178.577.687.479C19.146 20.115 22 16.379 22 11.974 22 6.465 17.535 2 12.026 2z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </div>
  <div
    class="c4"
  >
    <svg
      version="1.1"
      viewBox="0 0 86.60254037844386 100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M34.64101615137754 4.999999999999999Q43.30127018922193 0 51.96152422706632 4.999999999999999L77.94228634059948 20Q86.60254037844386 25 86.60254037844386 35L86.60254037844386 65Q86.60254037844386 75 77.94228634059948 80L51.96152422706632 95Q43.30127018922193 100 34.64101615137754 95L8.660254037844387 80Q0 75 0 65L0 35Q0 25 8.660254037844387 20Z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;
